### PR TITLE
Transaction Log Implementation

### PR DESCRIPTION
**Pull request by ItsMurumba**
This PR handle the basic setup of the Trasnaction Log Components. The tickets handled in this PR are:

- https://app.clickup.com/t/86bzqrr5p
- https://app.clickup.com/t/86bzqrqux

## Summary by CodeRabbit

- **New Features**
	- Enhanced the documentation with a "Tech Used" section and a note on local development setup.
	- Introduced a transaction log microfrontend with improved API functionality for transactions and channels.
	- Added new routing for the transaction log and sidebar applications.
	- Implemented new components for filtering and displaying transaction logs.

- **Bug Fixes**
	- Updated navigation links to ensure proper routing to the new transaction log.

- **Chores**
	- Added configuration files for ESLint, Babel, Prettier, Jest, and TypeScript to improve code quality and maintainability.
	- Introduced `.gitignore` and `.prettierignore` files to streamline version control and formatting processes.

### CODE REVIEW

File: packages/transaction-log/src/components/App.tsx [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+
+  const loadMore = () => {
+    setLimit(prevLimit => prevLimit + 20)
+  }
+
+  const filteredTransactions = transactions.filter(transaction => {
+    const searchTerm = searchQuery.toLowerCase()
+    return (
+      transaction.channelName?.toLowerCase().includes(searchTerm) ||
+      '' ||
```
Comment (marvinIsSacul) [inline packages/transaction-log/src/components/App.tsx:135]: What are the `|| ''` for?

File: packages/transaction-log/src/components/CustomFilters.tsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+            value={limit}
+            onChange={handleLimitChange}
+            fullWidth
+          >
+            <MenuItem value={10}>10</MenuItem>
+            <MenuItem value={20}>20</MenuItem>
+            <MenuItem value={50}>50</MenuItem>
+            <MenuItem value={100}>100</MenuItem>
+            <MenuItem value={200}>200</MenuItem>
+            <MenuItem value={500}>500</MenuItem>
```
Comment (marvinIsSacul) [inline packages/transaction-log/src/components/CustomFilters.tsx:135]: Dont you need key here? example:

`<MenuItem key={10} value={10}>10</MenuItem>`

File: packages/transaction-log/src/components/App.tsx [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+        transactions.map(async transaction => {
+          const channelName = await fetchChannelDetails(transaction.channelID)
+          const clientName = await fetchClientDetails(transaction.clientID)
+          return {...transaction, channelName, clientName}
+        })
+      )
+
+      setTransactions(transactionsWithChannelDetails)
+    } catch (error) {
+      console.error('Error fetching logs:', error)
```
Comment (marvinIsSacul) [inline packages/transaction-log/src/components/App.tsx:77]: Might need to handle errors and display them to user (dialogs, modals, etc.)

File: packages/transaction-log/src/components/App.tsx [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+    }
+  }, [status, channel, startDate, endDate, limit, reruns])
+
+  const fetchAvailableChannels = useCallback(async () => {
+    try {
+      const channels = await fetchChannels()
+      setChannels(channels)
+    } catch (error) {
+      console.error('Error fetching channels:', error)
```
Comment (marvinIsSacul) [inline packages/transaction-log/src/components/App.tsx:86]: Better error handling.

File: packages/transaction-log/src/components/App.tsx [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+  const fetchChannelDetails = async (channelID: string) => {
+    try {
+      const response = await fetchChannelById(channelID)
+      if (!response.ok) {
+        throw new Error('Network response was not ok')
+      }
+      const channel = await response.json()
+      return channel.name
+    } catch (error) {
+      console.error('Error fetching logs:', error)
```
Comment (marvinIsSacul) [inline packages/transaction-log/src/components/App.tsx:108]: Better error handling

File: packages/transaction-log/src/components/App.tsx [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+  const fetchClientDetails = async (clientID: string) => {
+    try {
+      const response = await fetchClientById(clientID)
+      if (!response.ok) {
+        throw new Error('Network response was not ok')
+      }
+      const client = await response.json()
+      return client.name
+    } catch (error) {
+      console.error('Error fetching logs:', error)
```
Comment (marvinIsSacul) [inline packages/transaction-log/src/components/App.tsx:122]: Better error handling.

File: packages/transaction-log/package.json [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    "@emotion/styled": "^11.11.5",
+    "@mui/icons-material": "^5.15.21",
+    "@mui/material": "^5.15.21",
+    "@mui/x-date-pickers": "^7.11.1",
+    "@types/jest": "^27.0.1",
+    "@types/react": "^17.0.19",
+    "@types/react-dom": "^17.0.9",
+    "@types/systemjs": "^6.1.1",
+    "@types/webpack-env": "^1.16.2",
+    "axios": "^1.7.2",
```
Comment (marvinIsSacul) [inline packages/transaction-log/package.json:62]: Do we need axios?

```suggestion

```

File: packages/transaction-log/src/components/App.tsx [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+} from '@mui/material'
+import CustomFilters from './CustomFilters'
+import BasicFilters from './BasicFilters'
+import TransactionLogTable from './TransactionLogTable'
+import {
+  fetchTransactions,
+  fetchChannelById,
+  fetchClientById,
+  fetchChannels
+} from '@jembi/openhim-core-api'
```
Comment (marvinIsSacul) [inline packages/transaction-log/src/components/App.tsx:20]: Would be nice to abstract these away to a service/api layer.

**Review by marvinIsSacul — CHANGES_REQUESTED** [review]
Looks good overall.

Just a couple of changes requested and the CI pipeline is failing.

**Comment by drizzentic** [comment]
@coderabbitai review

File: packages/transaction-log/src/components/common/app.main.component.tsx [hunk 9]
```diff
@@ -0,0 +1,339 @@
...
+  }, [])
+
+  // useEffect(() => {
+  //   const intervalId = setInterval(async () => {
+  //     const timestamp = new Date().toISOString()
+  //     await fetchTransactionLogs('2024-08-14T15:50:25+03:00') // Auto-polling with timestamp filter, without filterLimit and filterPage
+  //   }, 5000)
+
+  //   return () => clearInterval(intervalId) // Cleanup interval on component unmount
+  // }, [fetchTransactionLogs])
```
Comment (drizzentic) [inline packages/transaction-log/src/components/common/app.main.component.tsx:170]: Remove unused codes

File: packages/transaction-log/src/components/common/app.main.component.tsx [hunk 10]
```diff
@@ -0,0 +1,339 @@
...
+      transaction.request.host?.toLowerCase().includes(searchTerm) ||
+      '' ||
+      transaction.request.path?.toLowerCase().includes(searchTerm) ||
+      '' ||
+      transaction.request.params?.toLowerCase().includes(searchTerm) ||
+      '' ||
+      transaction.status?.toLowerCase().includes(searchTerm) ||
+      ''
+    )
+  })
```
Comment (drizzentic) [inline packages/transaction-log/src/components/common/app.main.component.tsx:225]:
```suggestion
const filteredTransactions = transactions.filter(transaction => {
  const searchTerm = searchQuery.toLowerCase();
  return [
    transaction.channelName,
    transaction.clientName,
    transaction.request.method,
    transaction.request.host,
    transaction.request.path,
    transaction.request.params,
    transaction.status
  ].some(field => field?.toLowerCase().includes(searchTerm));
});
```

File: packages/transaction-log/src/types/index.spec.ts [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+    expect(client).toHaveProperty('_id')
+    expect(client).toHaveProperty('name')
+    expect(typeof client._id).toBe('string')
+    expect(typeof client.name).toBe('string')
+  })
+
+  it('should fail if Client object is missing properties', () => {
+    // const client: Client = {
+    //   _id: '12345'
+    // }
```
Comment (drizzentic) [inline packages/transaction-log/src/types/index.spec.ts:19]: Remove commented code

File: packages/transaction-log/src/types/index.spec.ts [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+    expect(channel).toHaveProperty('_id')
+    expect(channel).toHaveProperty('name')
+    expect(typeof channel._id).toBe('string')
+    expect(typeof channel.name).toBe('string')
+  })
+
+  it('should fail if Channel object is missing properties', () => {
+    // const channel: Channel = {
+    //   name: 'General'
+    // }
```
Comment (drizzentic) [inline packages/transaction-log/src/types/index.spec.ts:39]: ditto

**Review by drizzentic — CHANGES_REQUESTED** [review]

File: packages/transaction-log/src/components/common/app.main.component.tsx (hunk 9 above)
Comment (drizzentic) [reply packages/transaction-log/src/components/common/app.main.component.tsx]: Seems we still have the commented code. Is it useful?

Comment (ItsMurumba) [reply packages/transaction-log/src/components/common/app.main.component.tsx]: I have removed them and do not exist in my local machine. @drizzentic

Comment (drizzentic) [inline packages/transaction-log/src/components/common/app.main.component.tsx]:
```suggestion
```

**Review by drizzentic — APPROVED** [review]

**Pull request merged**